### ISSUE TITLE
Support setting the primary key by users

### ISSUE THREAD

**Issue by shafreeck** [pingcap/tidb#8283]
## Feature Request

**Is your feature request related to a problem? Please describe:**

We are implementing a Redis layer based on TiKV, and we use a separate key(MetaKey) to store the meta data of a complex data structure (Ex. List), another key(DataKey) to store an element of the data structure.

When we commit a transaction which modifies both the MetaKey and DataKey, the DataKey is chosen as the primary key of the transaction for its sort order. The MetaKey will be committed asynchronously and at the same time, another request can be issued to the server and tries to read the MetaKey, the request triggers a backoff because of the commit before has not been completed.

**Describe the feature you'd like:**

Export an API to set the primary key of a transaction. This can be an option, use the first key as before if no primary key is supplied. 

**Describe alternatives you've considered:**

Glad to hear other suggestions and reasons if the feature request would not be accepted.

**Teachability, Documentation, Adoption, Migration Strategy:**

Maybe set txn.SetOption method can resolve this, like

```
txn.SetOption(txn.PrimaryKey, "this_is_the_primary_key")
```

**Comment by shenli** [comment]
@shafreeck Thanks for your feedback!
But I don't think it is a good idea. We choose primary key by order to avoid some kind of 'deadlock'.

**Comment by gregwebs** [comment]
@shafreeck can you ask this of TiKV? https://github.com/tikv/tikv

**Comment by siddontang** [comment]
@shafreeck 

If you really want to do it, you need to consider deadlock, and maybe you need to support wound-wait or wait-die to prevent it in TiKV.

**Comment by shafreeck** [comment]
@gregwebs It's a feature request for TiKV Go SDK which is in the TiDB project, Maybe it's more appropriate to discuss here?

**Comment by shafreeck** [comment]
@shenli @siddontang Sorry for not getting the idea about the 'deadlock'. What's the difference between first-key and non-first-key? Why the sort order affects the 'deadlock'?

I very appreciate if you could give me more details about this.

**Comment by disksing** [comment]
I don't think it could cause any deadlock problem. However, I think a better solution to this problem may be to set an option to disable the lazy commit, which means that the written transaction can only r turn after all regions have been committed.

**Comment by disksing** [comment]
`txn.SetOption` sounds good to me.
